sklearn/manifold/_barnes_hut_tsne.pyx

```diff
@@ -365,13 +365,20 @@ cdef int free_tree(Tree* tree) nogil:
     for i in range(3):
         cnt[i] = 0
     free_recursive(tree, tree.root_node, cnt)
-    free(tree.root_node)
-    free(tree)
     check = cnt[0] == tree.n_cells
     check &= cnt[2] == tree.n_points
+    free(tree)
     free(cnt)
     return check
 
+cdef void free_post_children(Node *node) nogil:
+    free(node.width)
+    free(node.left_edge)
+    free(node.center)
+    free(node.barycenter)
+    free(node.leaf_point_position)
+    free(node)
+
```

Comment:
Actually the test of the file does not follow pep8 blank lines so better consistent with the local file style in this PR.